core/api/make.cpp

```diff
@@ -345,8 +356,80 @@ namespace fc::api {
           OUTCOME_TRY(tipset, ts_load->load(tipset_key));
           return weight_calculator->calculateWeight(*tipset);
         }};
-    // TODO(turuslan): FIL-165 implement method
-    api->ClientFindData = {};
+
+    api->ClientFindData = waitCb<std::vector<QueryOffer>>(
+        [=](auto &&root_cid, auto &&piece_cid, auto &&cb) {
+          OUTCOME_CB(auto peers, market_discovery->getPeers(root_cid));
+          // if piece_cid is specified, remove peers without the piece_cid
+          if (piece_cid.has_value()) {
+            peers.erase(std::remove_if(peers.begin(),
+                                       peers.end(),
+                                       [&piece_cid](auto &peer) {
+                                         return peer.piece != *piece_cid;
+                                       }),
+                        peers.end());
+          }
+
+          std::mutex mutex;
```

Comment:
Use `AsyncAll<>`

Reply:
I don't see benefits from `AsyncAll<>` in this case. First, it fails if one of queries fails. Then, it cannot store call parameter that is being used here. The benefit might be in performance of `std::atomic` over `mutex`.  I don't think it is  relevant here so I prefer to leave it as is. The other option might be is to use the same approach as in 'AsyncAll' if performance is an issue. Third option is to rework `AsyncAll`, so it can be utilised here. Forth option is to add more wrappers for parameters and error handling and reuse 'AsyncAll'. I see the last option as undesirable since it reduces readability.